File: lect2/codes/decl_functions3.py
import numpy as np
def matmult(A = [], B = []):
    result = np.zeros((len(A[0]), len(B)))
    for i in range(len(A)):
        for j in range(len(B[0])):
            for k in range(len(A)):
                result[i][j] += A[i][k] * B[k][j]
    return (A, B, result)
if __name__ == "__main__":
    print(matmult([[1, 2], [2, 3]], [[3, 5],[4, 6]]))

File: lect2/codes/decl_functions3.m
clc;
function [res] = matmul(A, B)
  [m1, n1] = size(A);
  [m2, n2] = size(B);
  result = zeros([int32(n1), int32(m2)]);
  for i=1:m1
    for j=1:n2
      for k=1:m1
        result(i, j) += A(i, k) * B(k, j);
      end
    end
  end
  res = result;
end
disp(matmul([[1, 2]; [3, 4]], [[7, 5]; [4, 3]]));